app/api/answers/update_current.go

```diff
@@ -0,0 +1,79 @@
+package answers
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/go-chi/render"
+
+	"github.com/France-ioi/AlgoreaBackend/app/database"
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+func (srv *Service) updateCurrent(rw http.ResponseWriter, httpReq *http.Request) service.APIError {
+	var requestData updateCurrentRequest
+
+	var err error
+	if err = render.Bind(httpReq, &requestData); err != nil {
+		return service.ErrInvalidRequest(err)
+	}
+
+	user := srv.GetUser(httpReq)
+	if err = user.Load(); err == database.ErrUserNotFound {
+		return service.InsufficientAccessRightsError
+	}
+	service.MustNotBeError(err)
+
+	foundItemID, itemID, err := srv.Store.GroupAttempts().GetAttemptItemIDIfUserHasAccess(*requestData.AttemptID, user)
+	service.MustNotBeError(err)
+	if !foundItemID {
+		return service.InsufficientAccessRightsError
+	}
+
+	err = srv.Store.InTransaction(func(store *database.DataStore) error {
+		userAnswerStore := store.UserAnswers()
+		var currentAnswerID int64
+		currentAnswerID, err = userAnswerStore.GetOrCreateCurrentAnswer(user.UserID, itemID, requestData.AttemptID)
+		service.MustNotBeError(err)
+
+		columnsToUpdate := map[string]interface{}{
+			"sState":  *requestData.State,
+			"sAnswer": *requestData.Answer,
+		}
+		service.MustNotBeError(userAnswerStore.ByID(currentAnswerID).UpdateColumn(columnsToUpdate).Error())
+
+		service.MustNotBeError(store.UserItems().Where("idUser = ?", user.UserID).
+			Where("idItem = ?", itemID).
+			Where("idAttemptActive = ?", requestData.AttemptID).
+			UpdateColumn(columnsToUpdate).Error())
+
+		return nil
+	})
+	service.MustNotBeError(err)
+
+	service.MustNotBeError(render.Render(rw, httpReq, service.UpdateSuccess(nil)))
+	return service.NoError
+}
+
+type updateCurrentRequest struct {
```

Comment:
I don't get why they are optional while you do not accept them not being set. And you do not use types.* at all?

Reply:
they are not optionale, they are pointers just to be able to compare them with nil. Changed it to use types.* (not sure if it is better or not)

Reply:
I quite like/prefer the `return types.Validate([]string{"attempt_id", "answer", "state"}, &requestData.AttemptID, ...` over the long check.
The `*requestData.State` -> ` requestData.State.String.Value,` is not necessarily better.
The `Where("idAttemptActive = ?", requestData.AttemptID).` -> `Where("idAttemptActive = ?", requestData.AttemptID.Value).` is probably worse.
The `AttemptID *int64  ` -> `AttemptID types.RequiredInt64` ... as we are expecting it to be set, I quite prefer the "RequiredInt64" over the optional.
Actually, if these types are used only for input validation, maybe it would clearer to have the package called something like "input" or "inputvalidation or "???", to have something like `AttemptID input.RequiredInt64 `. 

In summary, here, I find it a bit better.